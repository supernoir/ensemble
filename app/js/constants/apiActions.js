const API_ACTIONS = {
	GET   : 'get',
	POST  : 'post',
	PUT   : 'put',
	DELETE: 'delete'
};


export default API_ACTIONS;