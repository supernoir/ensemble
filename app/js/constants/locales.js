const LOCALES = [
	{ locale: 'en_US', code: 'us', name: 'lang.en_US' },
	{ locale: 'es_ES', code: 'es', name: 'lang.es_ES' }
];

export default LOCALES;