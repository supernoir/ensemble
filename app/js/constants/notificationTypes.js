/**
 * Notification Types Constant
 */
export const notificationTypes = {
	INFO   : 'info',
	WARN   : 'warning',
	ERROR  : 'error',
	SUCCESS: 'success'
};
