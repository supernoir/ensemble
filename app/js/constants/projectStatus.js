/**
 * Project Status Constant
 */
export const projectStatus = {
	DRAFT      : 'draft',
	IN_PROGRESS: 'inprogress',
	IN_REVIEW  : 'inreview',
	EDITING    : 'editing',
	PUBLISHED  : 'published'
};

export const projectStatusColormapping = {
	draft      	: 'grey',
	inprogress: 'blue',
	inreview  	: 'red',
	editing    	: 'yellow',
	published  	: 'green'
};