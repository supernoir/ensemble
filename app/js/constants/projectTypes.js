export const projectTypes = {
	SHORTSTORY: 'shortstory',
	SCRIPT    : 'script',
	NOVEL     : 'novel',
	ESSAY     : 'essay',
	poem      : 'poem'
};
