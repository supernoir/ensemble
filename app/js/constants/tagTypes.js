export const tagTypes = {
	// Specific tag for only one project
	PROJECT      : 'project',
	// User-based tag
	USER         : 'user',
	// Global tag for all users
	GLOBAL       : 'global',
	// tag represents project state
	PROJECT_STATE: 'project_state',
	// tag represents project type
	PROJECT_TYPE : 'project_type'
};
