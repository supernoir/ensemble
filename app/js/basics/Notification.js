import React from 'react';
import { Message } from 'semantic-ui-react';
import { notificationTypes } from '../constants/notificationTypes';
import PropTypes from 'prop-types';

/**
 * Class Notification
 * handles displaying notifications in the app
 */
export default class Notification extends React.Component {
	constructor() {
		super();
		this.state = {
			visible: true
		};
	}

	parseNotificationType = type => {
		switch (type) {
			case notificationTypes.INFO:
				return notificationTypes.INFO;
			case notificationTypes.WARN:
				return notificationTypes.WARN;
			case notificationTypes.ERROR:
				return notificationTypes.ERROR;
			case notificationTypes.SUCCESS:
				return notificationTypes.SUCCESS;
			default:
				return '';
		}
	};

	handleDismiss = () => {
		this.setState({ visible: false });

		setTimeout(() => {
			this.setState({ visible: true });
		}, 2000);
	};

	render() {
		return (
			<Message className={this.props.type} onDismiss={this.handleDismiss}>
				<Message.Header>{this.props.content}</Message.Header>
				<p>{this.props.description}</p>
			</Message>
		);
	}
}

Notification.propTypes = {
	type       : PropTypes.string,
	content    : PropTypes.string,
	description: PropTypes.string
};