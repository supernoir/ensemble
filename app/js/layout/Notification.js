import React from 'react';
import { Message } from 'semantic-ui-react';
import { notificationTypes } from '../constants/notificationTypes';

export default class Notification extends React.Component {
	constructor() {
		super();
		this.state = {
			visible: true
		};
	}

	parseNotificationType = type => {
		switch (type) {
			case notificationTypes.INFO:
				return notificationTypes.INFO;
			case notificationTypes.WARN:
				return notificationTypes.WARN;
			case notificationTypes.ERROR:
				return notificationTypes.ERROR;
			case notificationTypes.SUCCESS:
				return notificationTypes.SUCCESS;
			default:
				return '';
		}
	};

	handleDismiss = () => {
		this.setState({ visible: false });

		setTimeout(() => {
			this.setState({ visible: true });
		}, 2000);
	};

	render() {
		return (
			<Message className={this.props.type} onDismiss={this.handleDismiss}>
				<Message.Header>{this.props.content}</Message.Header>
				<p>{this.props.description}</p>
			</Message>
		);
	}
}
