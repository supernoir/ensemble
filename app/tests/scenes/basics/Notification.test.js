import React from 'react';
import { shallow } from 'enzyme';
import { notificationTypes } from '../../../js/constants/notificationTypes';

const testData = {
	type       : notificationTypes.ERROR,
	content    : 'A test error occurred',
	description: 'Something went wrong'
};

// Import Components
import Notification from '../../../js/basics/Notification';

test('Notification -- Snapshot Test', () => {
	const wrapper = shallow(<Notification
		type={testData.type}
		content={testData.content}
		description={testData.description}
	/>
	);
	expect(wrapper).toMatchSnapshot();
});

test('Notification -- Types Tests', () => {
	const NotificationWithError = shallow(<Notification type={notificationTypes.ERROR} />);
	expect(NotificationWithError.find('.error')).toHaveLength(1);
	const NotificationWithWarning = shallow(<Notification type={notificationTypes.WARN} />);
	expect(NotificationWithWarning.find('.warning')).toHaveLength(1);
	const NotificationWithSuccess = shallow(<Notification type={notificationTypes.SUCCESS} />);
	expect(NotificationWithSuccess.find('.success')).toHaveLength(1);
	const NotificationWithInfo = shallow(<Notification type={notificationTypes.INFO} />);
	expect(NotificationWithInfo.find('.info')).toHaveLength(1);
});
