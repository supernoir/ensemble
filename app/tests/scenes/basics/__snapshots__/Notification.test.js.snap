// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification -- Snapshot Test 1`] = `
<Message
  className="error"
  onDismiss={[Function]}
>
  <MessageHeader>
    A test error occurred
  </MessageHeader>
  <p>
    Something went wrong
  </p>
</Message>
`;
